File: lab5/Tour.java
package ua.lviv.iot.lab3;

public abstract class Tour {
    private String name;
    private int duration;
    private double price;
    private String country;
    private TourType tourType;

    public Tour() {
        }

    public Tour(final String name, final int duration, final double price, final String country) {
        this.name = name;
        this.duration = duration;
        this.price = price;
        this.country = country;
        }

    public abstract TourType getTourType();

    public Tour(final String name, final TourType tourType, final int duration) {
        this.name = name;
        this.tourType = tourType;
        this.duration = duration;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }
}

File: lab11/Tour.py
from enum import Enum


class TourType(Enum):
    FAMILY = 1
    SPORT = 2
    EXTREME = 3


class Tour:
    type = None
    duration = 0
    price = 0.0
    country = None

    def __init__(self, duration, price, country):
        self.duration = duration
        self.price = price
        self.country = country

    def __str__(self):
        return self.type.name + " " + str(self.duration) + " " + str(self.price) + " " + self.country
